lib/Doctrine/DBAL/Exception/ConstraintViolationException.php

```diff
@@ -28,4 +28,17 @@
  */
 class ConstraintViolationException extends ServerException
 {
+	/**
+	 * Get the name of affected column.
+	 *
+	 * @return string|NULL
+	 */
+	public function getAffectedColumnName()
+	{
+		if (preg_match_all('/^DETAIL\:\s+Key\s+\((\w+)\)/im', $this->getMessage(), $matches)) {
```

Comment:
Hey @deeky666,
so I have collected error messages from `PostgreSQL`, `MySQL`, `SQLite` and `SQL Server`. They are all different and I cannot reliably get column name, they tend to use index name. How shall I proceed?

Is this PR bad idea?